ls1intum/Artemis
# [Enhancement] Result view

### Checklist

### Motivation and Context



Improve the visual presentation of feedbacks.

### Description


- Feedback items are now displayed based on their type (Test Case / SCA Issue / Manual Feedback) and their credits.
- Feedback items show their credits, if they have some.
- Students do not see the test case names and have no distinction between automatic and manual Feedback. (All is labeled 'Feedback')
- Instructors see test case names, positive test cases and have a distinction between test case and manual feedback. (Labeled either Test Case or Tutor)
- A score bar is added on top of the feedback, showing achieved points and deductions
- A 'Preliminary' badge is shown if the result is of type Automatic, because of potential hidden test cases.

### Steps for Testing


Check the feedback view for different exercises and both as student and instructor. The feedback items should be shown as described above.

1. Test for a Programming Exercise without manual feedback.
2. Test for a Programming Exercise with manual feedback.
3. Test for the other exercise types that the results and feedbacks still display correctly.

### Test Coverage


**result-detail.component.ts**: 94.44% stmts

### Screenshots


**Student View**
<img width="814" alt="Bildschirmfoto 2020-10-26 um 02 23 14" src="https://user-images.githubusercontent.com/13920539/97124899-40ecda80-1732-11eb-9c54-bc612bff422d.png">
**Instructor View**
<img width="816" alt="Bildschirmfoto 2020-10-26 um 01 55 33" src="https://user-images.githubusercontent.com/13920539/97124897-3fbbad80-1732-11eb-86f6-abf68e5f1a63.png">
